### DISCUSSION TITLE
micropython can support Apple homekit？

### DISCUSSION THREAD

**Discussion by coffice12** [micropython/micropython#15560] [General]

**Comment by dpgeorge** [comment]
It sounds like this is a feature request.  Please can you provide more detail so progress can be made.

**Comment by ukd1** [comment]
@dpgeorge seems likely this should be an external library, not core? but at least my request would be to have some kind of library (core or otherwise) that supported HomeKit protocol. Python library examples are here:

* https://github.com/ikalchev/HAP-python - issue tracking it ikalchev/HAP-python#286
* https://github.com/jlusiardi/homekit_python
* https://github.com/condemil/pyhap

I may port one at some point, but tbh I'm not a Python person really.

**Comment by MarByteBeep** [comment]
What's the current status on this one? It would be really cool if for Apple Homekit would be supported on Micropython driven devices, for obvious reasons. I fail to believe that my smart light bulb is actually smarter than for instance a Raspberry Pico W :)

**Comment by jonnor** [comment]
Some work has been done in https://github.com/katadelos/micropython-esp32-homekit
In general, this seems to be doable with either a Python or C module - so it does not have to happen in core MicroPython repository. Proposing to move this issue to a discussion.

**Comment by tannewt** [comment]
I've started a Python Matter implementation: https://github.com/adafruit/circuitmatter Matter also works with Apple Home. Right now it is for CPython but I plan on porting it to CircuitPython too. MicroPython shouldn't be hard either. I hope to collaborate on it.
